secant/Features/BackupFlow/RecoveryPhraseDisplayStore.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  RecoveryPhraseDisplayStore.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/26/21.
+//
+
+import Foundation
+import ComposableArchitecture
+import UIKit
+
+enum RecoveryPhraseError: Error {
+    /// This error is thrown then the Recovery Phrase can't be generated
+    case unableToGeneratePhrase
+}
+
+struct BackupPhraseEnvironment {
+    var mainQueue: AnySchedulerOf<DispatchQueue>
+    var newPhrase: () -> Effect<RecoveryPhrase, RecoveryPhraseError>
```

Comment:
Should newPhrase always return the same phrase?  Since there are no inputs here, whenever you make the environment, and provide this function, you are left with an environment that always produces the same phrase when this function is called.  I'm not certain if that is the desired behavior or not, but if this function took the inputs of the phase as the inputs of this function, I believe this would be more dynamic and a newPhrase could be generated from any [String].  Again, not entirely sure if this is what you want, just bringing up for discussion sake.

Reply:
Although this is only there for mock up purposes, the behavior of the real feature should generate a random mnemonic phrase with the given library **only once** and save it to the keychain regardless of the validation results. 

If the user wants to get rid of the phrase, that should require an explicit action (nuking the wallet). which means passing the validation test, getting to the profile/settings screen and nuking it.